src/cloudant/result.py

```diff
@@ -360,10 +360,11 @@ def __getitem__(self, key):
 
         :returns: Rows data in JSON format
         """
-        if 'skip' in self.options or 'skip' in self._ref.keys():
+        refkeys = set(self._ref.keys())
+        if 'skip' in self.options or 'skip' in refkeys:
             msg = 'Cannot use skip parameter with QueryResult slicing.'
             raise CloudantArgumentError(msg)
-        if 'limit' in self.options or 'limit' in self._ref.keys():
+        if 'limit' in self.options or 'limit' in refkeys:
```

Comment:
in py3, keys() returns an iterator which would then have to be forced into list form for "in" testing.  Rather than do that twice back to back, I chose to do it once and make it a set to improve performance, ie. fewer list constructions and an O(1) vs. O(n) membership test.